test/circuit_breaker_test.rb

```diff
@@ -6,72 +6,134 @@ class TestCircuitBreaker < Minitest::Test
+      open_circuit!(resource, error_count: 3)
+    rescue Semian::OpenCircuitError
     end
-    assert_match(/State transition from closed to open/, @strio.string)
   end
 
   def test_last_error_message_is_logged_when_circuit_opens
-    open_circuit!
-    assert_match(/last_error_message="some error message"/, @strio.string)
+    resource = Semian.register(
+      :circuit_is_open_last_error,
+      bulkhead: false,
+      exceptions: [SomeError],
+      error_threshold: 1,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+    assert_log_message_match(/last_error_message="some error message 0"/, level: :info) do
+      3.times do |i|
+        resource.acquire { raise SomeError, "some error message #{i}" }
+      rescue SomeError
+      end
+    rescue Semian::OpenCircuitError
+    end
   end
 
   def test_after_error_threshold_the_circuit_is_open
-    open_circuit!
-    assert_circuit_opened
+    resource = Semian.register(
+      :circuit_is_open_after_error_threshold,
+      bulkhead: false,
+      exceptions: [SomeError],
+      error_threshold: 3,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+    open_circuit!(resource, error_count: 3)
+    assert_circuit_opened(resource)
   end
 
   def test_after_error_timeout_is_elapsed_requests_are_attempted_again
-    half_open_cicuit!
-    assert_circuit_closed
+    resource = Semian.register(
+      :circuit_is_open_after_error_timeout_try_again,
+      bulkhead: false,
+      exceptions: [SomeError],
+      error_threshold: 3,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+    half_open_cicuit!(resource, error_count: 3, backwards_time_travel: 10)
+    assert_circuit_closed(resource)
   end
 
   def test_until_success_threshold_is_reached_a_single_error_will_reopen_the_circuit
-    half_open_cicuit!
-    trigger_error!
-    assert_circuit_opened
-    assert_match(/State transition from open to half_open/, @strio.string)
+    resource = Semian.register(
+      :circuit_is_open_success_single_error,
+      bulkhead: false,
+      exceptions: [SomeError],
+      error_threshold: 3,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
+    half_open_cicuit!(resource, error_count: 3, backwards_time_travel: 10)
+    expected = /State transition from open to half_open.*State transition from half_open to open/m
+    assert_log_message_match(expected) do
+      trigger_error!(resource)
+    end
+    assert_circuit_opened(resource)
   end
 
   def test_once_success_threshold_is_reached_only_error_threshold_will_open_the_circuit_again
-    half_open_cicuit!
-    assert_circuit_closed
-    trigger_error!
-    assert_circuit_closed
-    trigger_error!
-    assert_circuit_opened
+    resource = Semian.register(
+      :circuit_once_success_threshold_is_reached_only_error_threshold_will_open_the_circuit_again,
+      bulkhead: false,
+      exceptions: [SomeError],
+      error_threshold: 3,
+      error_timeout: 5,
+      success_threshold: 1,
+    )
```

Comment:
looks like `Semian.register` could be extracted to the factory method, and hide redundant per test example params